### PR TITLE
Automatically refresh extension when profile config is updated

### PR DESCRIPTION
**Pull request by kristinochka**
## Proposed changes

Watch for global and project profiles to be created, updated, or deleted and automatically refresh ZE extension when changes happen. 

## Release Notes

Milestone:
 Automatically refresh extension when project or global profile is created, updated, or deleted.

Changelog:
https://github.com/zowe/vscode-extension-for-zowe/issues/1760

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [x] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [ ] ~I have added necessary documentation (if appropriate)~
- [ ] ~Any PR dependencies have been merged and published (if appropriate)~

### CODE REVIEW

**Comment by kristinochka** [comment]
https://user-images.githubusercontent.com/14590297/179259091-be1f8117-16f7-4dba-b913-8e5ca19b45cf.mov

**Comment by kristinochka** [comment]
I have disabled this in Theia because there are some issues. One issue is file changes are not picked up in .zowe folder, I suspect this is related to Theia implementation of the watcher.

**Review by rudyflores — APPROVED** [review]
LGTM! Thanks for these changes @kristinochka tested this in Che as well and seems to be working as intended!

**Comment by JillieBeanSim** [comment]
> I have disabled this in Theia because there are some issues. One issue is file changes are not picked up in .zowe folder, I suspect this is related to Theia implementation of the watcher.

Hey @kristinochka what version of Theia did you test against and are there any open issues for Theia to pick up this support for the VSC APIs for watching files?  If there are no open issues should we create one?

**Comment by kristinochka** [comment]
I have only tested this in our current version of Eclipse Che. I took a quick look at Theia issues, and there are many related to watchers: https://github.com/eclipse-theia/theia/issues?page=1&q=is%3Aissue+is%3Aopen+watcher, though I have not debugged this Theia issue in detail to know is any of them are related.

File: packages/zowe-explorer/src/extension.ts [hunk 1]
```diff
@@ -344,9 +344,65 @@ export async function activate(context: vscode.ExtensionContext): Promise<ZoweEx
     ZoweExplorerExtender.createInstance(datasetProvider, ussFileProvider, jobsProvider);
 
     await SettingsConfig.standardizeSettings();
+    await watchConfigProfile(context);
     return ZoweExplorerApiRegister.getInstance();
 }
 
+async function watchConfigProfile(context: vscode.ExtensionContext) {
+    if (globals.ISTHEIA) {
+        return undefined;
```
Comment (JillieBeanSim) [inline packages/zowe-explorer/src/extension.ts:353]: If ISTHEIA do we have a route to take, I see the pop up message with reload button was removed.  Should we keep that functionality for Theia environment?

Comment (kristinochka) [reply packages/zowe-explorer/src/extension.ts:353]: Yes, it should still be there, good catch.

**Comment by JillieBeanSim** [comment]
@kristinochka seeing this linting error
```ERROR: 678:11  member-access  The class method 'propmtToRefreshForProfiles' must be marked either 'private', 'public', or 'protected'```

File: packages/zowe-explorer/src/Profiles.ts [hunk 2]
```diff
@@ -1313,6 +1303,23 @@ export class Profiles extends ProfilesCache {
         await vscode.window.showTextDocument(document);
     }
 
+    private async propmtToRefreshForProfiles(rootPath: string) {
```
Comment (JillieBeanSim) [inline packages/zowe-explorer/src/Profiles.ts:1306]: could the spelling be corrected here and the call to it `promptToRefreshForProfiles`

**Review by JillieBeanSim — CHANGES_REQUESTED** [review]
I am seeing an issue with having project level and global level configs wanting to use each others secure credentials. I do not see this issue in the current state of master branch.

I cleared any local cached Zowe instances in my local keystore and in testing this branch only one instance is being written back even though they are different and should be 2 instances in the keystore.

**Comment by JillieBeanSim** [comment]
@kristinochka all looks good on local and in theia.  Could you address spelling comment and run `yarn workspace vscode-extension-for-zowe vscode:prepublish` to pick up the changes from pop up message in theia localization.  After that commit I can approve. Thanks

File: packages/zowe-explorer/src/Profiles.ts (hunk 2 above)
Comment (kristinochka) [reply packages/zowe-explorer/src/Profiles.ts]: Done.

**Review by JillieBeanSim — APPROVED** [review]
LGTM! thanks @kristinochka for your work on this much needed feature

**Review by rudyflores — APPROVED** [review]

**Pull request merged**